src/textual/reactive.py

```diff
@@ -50,6 +53,108 @@ class TooManyComputesError(ReactiveError):
     """Raised when an attribute has public and private compute methods."""
 
 
+class WatchDecorator(Generic[WatchMethodType]):
+    """Watch decorator.
+
+    Decorate a method to make it a watcher.
+
+    """
+
+    def __init__(self, reactive: Reactive | None = None) -> None:
+        self._reactive = reactive
+
+    @overload
+    def __call__(self, *, init: bool = True) -> WatchDecorator[WatchMethodType]: ...
+
+    @overload
+    def __call__(self, method: WatchMethodType) -> WatchMethodType: ...
+
+    def __call__(
+        self, method: WatchMethodType | None = None, *, init: bool = True
+    ) -> WatchMethodType | WatchDecorator[WatchMethodType]:
+        _rich_traceback_omit = True
+        if method is None:
+            return self
+        assert self._reactive is not None
+        assert hasattr(method, "__name__")
+        if not method.__name__.startswith("watch_"):
```

Comment:
I'm a bit confused. If the decorated method has a name starting with `_watch`, we don't do the duplicate method checking / raise the error and instead just override the previous decorated watch method?
Similar comments apply to the compute and validate decorators.

In other words, if `test` is a reactive, then the first option below raises an error but the 2nd one doesn't?

```py
class MyApp(App[None]):
    test = reactive(42)

    @test.watch
    def _(self):
        pass

    @test.watch
    def _2(self):
        pass
```

```py
class MyApp(App[None]):
    test = reactive(42)

    @test.watch
    def _(self):
        pass

    @test.watch
    def watch_2(self):
        pass
```

Reply:
This is to exclude match methods defined via the naming convention. Without it, a decorated watch method may be called twice.

It may be better to prohibit decorating a watch method. What do you think?

Reply:
I think decorating a watch method will look confusing but if we end up dropping the naming convention either way, is it worth creating a special rule now and then remove it?